src/Stratis.Features.FederatedPeg.IntegrationTests/NodeInitialisationTests.cs

```diff
@@ -0,0 +1,194 @@
+
+        private readonly (Script payToMultiSig, BitcoinAddress sidechainMultisigAddress, BitcoinAddress
+            mainchainMultisigAddress) scriptAndAddresses;
+
+
+        public NodeInitialisationTests()
+        {
+            this.sidechainNetwork = (CirrusRegTest)CirrusNetwork.NetworksSelector.Regtest();
+            this.mainNetwork = Networks.Stratis.Regtest();
+            var pubKeysByMnemonic = this.sidechainNetwork.FederationMnemonics.ToDictionary(m => m, m => m.DeriveExtKey().PrivateKey.PubKey);
+            this.scriptAndAddresses = FederatedPegTestHelper.GenerateScriptAndAddresses(this.mainNetwork, this.sidechainNetwork, 2, pubKeysByMnemonic);
+        }
+
+        [Fact]
+        public void SidechainUserStarts()
+        {
+            using (SidechainNodeBuilder nodeBuilder = SidechainNodeBuilder.CreateSidechainNodeBuilder(this))
+            {
+                CoreNode user = nodeBuilder.CreateSidechainNode(this.sidechainNetwork);
+
+                user.Start();
+
+                Assert.Equal(CoreNodeState.Running, user.State);
+            }
+        }
+
+
+        [Fact]
+        public void SidechainMinerStarts()
+        {
+            using (SidechainNodeBuilder nodeBuilder = SidechainNodeBuilder.CreateSidechainNodeBuilder(this))
+            {
+                Key federationKey = new Key();
+
+                CoreNode miner = nodeBuilder.CreateSidechainMinerNode(this.sidechainNetwork, this.mainNetwork, federationKey);
+
+                miner.Start();
+
+                Assert.Equal(CoreNodeState.Running, miner.State);
+            }
+        }
+
+        [Fact]
+        public void SidechainGatewayStarts()
+        {
+            using (SidechainNodeBuilder nodeBuilder = SidechainNodeBuilder.CreateSidechainNodeBuilder(this))
+            {
+                Key federationKey = new Key();
+
+                CoreNode gateway = nodeBuilder.CreateSidechainFederationNode(this.sidechainNetwork, this.mainNetwork, federationKey);
+                gateway.AppendToConfig("sidechain=1");
+                gateway.AppendToConfig($"redeemscript={this.scriptAndAddresses.payToMultiSig.ToString()}");
+                gateway.AppendToConfig($"publickey={this.sidechainNetwork.FederationMnemonics[0].DeriveExtKey().PrivateKey.PubKey}");
+                gateway.AppendToConfig("federationips=0.0.0.0,0.0.0.1"); // Placeholders
+
+                gateway.Start();
+
+                Assert.Equal(CoreNodeState.Running, gateway.State);
+            }
+        }
+
+        [Fact]
+        public void MainChainGatewayStarts()
+        {
+            using (SidechainNodeBuilder nodeBuilder = SidechainNodeBuilder.CreateSidechainNodeBuilder(this))
+            {
+                CoreNode gateway = nodeBuilder.CreateMainChainFederationNode(this.mainNetwork, this.sidechainNetwork);
+                gateway.AppendToConfig("mainchain=1");
+                gateway.AppendToConfig($"redeemscript={this.scriptAndAddresses.payToMultiSig.ToString()}");
```

Comment:
Can't we perhaps move all of this config to a method or something? It looks like the same on all the tests, apart from the miainchain / sidechain config parameter. Something like `gateway.ConfigureMainchain` or `gateway.ConfigureSidechain`

Reply:
We could...

I kind of like it like this because it's very explicit what's being given to the node each time (always the least possible parameters), it's very easy to come to this test and see the required params